homeassistant/components/media_player/cast.py

```diff
@@ -388,8 +390,36 @@ def new_cast_status(self, cast_status):
 
     def new_media_status(self, media_status):
         """Handle updates of the media status."""
+        # Only use media position for playing/paused,
+        # and for normal playback rate
+        if (media_status is None or
+                abs(media_status.playback_rate - 1) > 0.01 or
+                not (media_status.player_is_playing or
+                     media_status.player_is_paused)):
+            self.media_status_position = None
+            self.media_status_position_received = None
+        else:
+            # Avoid unnecessary state attribute updates if player_state and
+            # calculated position stay the same
+            now = dt_util.utcnow()
+            do_update = \
+                (self.media_status is None or
+                 self.media_status_position is None or
+                 self.media_status.player_state != media_status.player_state)
+            if not do_update:
+                if media_status.player_is_playing:
+                    elapsed = now - self.media_status_position_received
+                    do_update = abs(media_status.current_time -
+                                    (self.media_status_position +
+                                     elapsed.total_seconds())) > 1
```

Comment:
@dersger That shouldn't be an issue since the `MEDIA_STATUS` package always contains the `currentTime` attribute. At least that's the case with all apps I just tested: Spotify, YouTube and Netflix.

However, there's a really weird thing going on: With the packets we receive when we manually poll the media status with `GET_STATUS`. We don't get the *actual* `currentTime` in the track, but the current time from the last callback! So for example, if I pause and play a track at 30 seconds, we will first receive `MEDIA_STATUS` packages with `currentTime = 30` from the callback. But even 30 and 60 seconds later when manually *poll* the device, `currentTime` in the packet will *still* be 30 - even though it should be 60/90 by now.

As there's no information from the cast packages about relative to which timestamp `currentTime` is measured, I think we can't really solve this problem :( So I'd say maybe remove the half-polling feature again?

Reply:
ugggh. Can we distinguish between polling and non-polling updates?

Reply:
With the current pychromecast library, AFAIK not. We *might* be able to create two MediaControllers on two different cast socket "channels", and use one of them for Push-Updates and the other one for polling updates. But that would probably just end up being a lot of spaghetti code :(